### DISCUSSION TITLE
What is the proper #include for the InputKeyboardSourceEventArgs ?

### DISCUSSION THREAD

**Discussion by shelllet** [microsoft/microsoft-ui-xaml#10096] [Q&A] [answered]
i want to hook the  `escape` key for contentdialog, so i found the flowing code:

````
class RespondToKeyDown
{
    InputKeyboardSource myInputKeyboardSource;

    public RespondToKeyDown(Microsoft.UI.Content.ContentIsland island)
    {
        myInputKeyboardSource = InputKeyboardSource.GetForIsland(island);
        myInputKeyboardSource.KeyDown += OnKeyDown;
    }

    void OnKeyDown(
        InputKeyboardSource sender,
        InputKeyboardSourceEventArgs args)
    {
        if (args.VirtualKey == Windows.System.VirtualKey.Escape)
        {
            System.Diagnostics.Debug.WriteLine("Escape key was pressed.");
            CancelCurrentOperation();
            args.Handled = true;
        }
    }
}
````
but i can't find the the type of `InputKeyboardSourceEventArgs` ,
how did i find where is the type of `InputKeyboardSourceEventArgs` declared in windowsappsdk winrt/c++?

and how can i do to hook the `escape` key for contentdialog?

**Comment by DarranRowe** [comment, marked as answer]
Well, the most important thing to note here is that the sample is wrong in regards to the InputKeyboardSourceEventArgs type.

The [documentation for the KeyDown event](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.input.inputkeyboardsource.keydown?view=windows-app-sdk-1.6#microsoft-ui-input-inputkeyboardsource-keydown) shows that what it has is a Microsoft.UI.Input.KeyEventArgs parameter. This is also reflected in the C++/WinRT headers.

```
int32_t __stdcall add_KeyDown(void* handler, winrt::event_token* token) noexcept final try
        {
            zero_abi<winrt::event_token>(token);
            typename D::abi_guard guard(this->shim());
            *token = detach_from<winrt::event_token>(this->shim().KeyDown(*reinterpret_cast<winrt::Windows::Foundation::TypedEventHandler<winrt::Microsoft::UI::Input::InputKeyboardSource, winrt::Microsoft::UI::Input::KeyEventArgs> const*>(&handler)));
            return 0;
        }
        catch (...) { return to_hresult(); }
```
Notice the reinterpret_cast's type that it is converting to.
